sormas-e2e-tests/src/test/java/org/sormas/e2etests/steps/api/ImmunizationSteps.java

```diff
@@ -0,0 +1,89 @@
+/*
+ * SORMAS® - Surveillance Outbreak Response Management & Analysis System
+ * Copyright © 2016-2021 Helmholtz-Zentrum für Infektionsforschung GmbH (HZI)
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program. If not, see <https://www.gnu.org/licenses/>.
+ */
+package org.sormas.e2etests.steps.api;
+
+import com.github.javafaker.Faker;
+import com.google.common.truth.Truth;
+import cucumber.api.java8.En;
+import io.restassured.response.Response;
+import java.util.Calendar;
+import java.util.UUID;
+import javax.inject.Inject;
+import org.sormas.e2etests.enums.*;
+import org.sormas.e2etests.enums.immunizations.ImmunizationManagementStatusValues;
+import org.sormas.e2etests.enums.immunizations.ImmunizationStatusValues;
+import org.sormas.e2etests.enums.immunizations.MeansOfImmunizationValues;
+import org.sormas.e2etests.helpers.api.ImmunizationHelper;
+import org.sormas.e2etests.pojo.api.Immunization;
+import org.sormas.e2etests.pojo.api.Person;
+import org.sormas.e2etests.state.ApiState;
+
+public class ImmunizationSteps implements En {
+
+  @Inject
+  public ImmunizationSteps(ImmunizationHelper immunizationHelper, ApiState apiState, Faker faker) {
+
+    When(
+        "API: I create {int} new immunizations for last created person",
+        (Integer numberOfImmunizations) -> {
+          String lastCreatedUserUUID = apiState.getCreatedPerson().getUuid();
+          Person person = Person.builder().uuid(lastCreatedUserUUID).build();
+          int immunizationAttempts = 1;
+          while (immunizationAttempts <= numberOfImmunizations) {
```

Comment:
Any reason to not use a for loop here?

Reply:
Foreach cannot be used and I considered while more simpler for our needs :D if we the for loop is more performant, we can change it, sure